### ISSUE TITLE
how to get grads for means2d from rasterization fn?

### ISSUE THREAD

**Issue by hardikdava** [nerfstudio-project/gsplat#467]
hello, I want to get grads of means2d from rasterization function. But I am getting None value. Is there anyway to get it without strategy?

**Comment by rylynchen** [comment]
+1, I met the same problem. Does anyone have the solution?

**Comment by yzslab** [comment]
Try adding a `means2d.retain_grad()` before your `loss.backward()`.

**Comment by hardikdava** [comment]
@yzslab Thanks for the comments. I have already tried it but did not work.

**Comment by yzslab** [comment]
> @yzslab Thanks for the comments. I have already tried it but did not work.

Make sure you are accessing exactly the same `means2d` tensor as the one returned by `rasterization`.
If you apply any additional operations, i.e. `means2d = some_func(means2d)`, then it is not the same one and the `grad` will be `None`.

**Comment by hardikdava** [comment]
@yzslab Thanks for the help. It is working now. So solution will be as following.

- rasterize
- retain grad
- do other ops

**Comment by rylynchen** [comment]
@yzslab Thanks for your sample `means2d = some_func(means2d)`
I just make the mistake like this, and now it work after fix it. 
Thank you !

**Comment by shawnFuu** [comment]
> > [@yzslab](https://github.com/yzslab) Thanks for the comments. I have already tried it but did not work.
> 
> Make sure you are accessing exactly the same `means2d` tensor as the one returned by `rasterization`. If you apply any additional operations, i.e. `means2d = some_func(means2d)`, then it is not the same one and the `grad` will be `None`.

I must give the additional advice. If you do `means2d = means2d.squeeze(0)`, then the means2d.grad=None! I have encountered this issue (orz
